Library/Homebrew/extend/os/mac/software_spec.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+class SoftwareSpec
+  def uses_from_macos(deps, **args)
+    depends_on(deps) if add_mac_dependency?(args)
+  end
+
+  private
+
+  def add_mac_dependency?(args)
+    args.each { |key, version| args[key] = OS::Mac::Version.from_symbol(version) }
+
+    return false if args[:after] && OS::Mac.version < args[:after]
+
+    return false if args[:before] && OS::Mac.version >= args[:before]
+
+    true
+  end
```

Comment:
From my understanding, `uses_from_macos "foo"` (i.e. without version args)  should not add `foo` as dependency on macOS. However, the current logic does the opposite.